### PR TITLE
Created basic Client and Server class.

### PR DESCRIPTION
**Pull request by LMiljak**
This branch copy & pasted files from the other networking branch (#16), but because merging it with the master was an incredible difficulty, I decided that creating a new branch from master and then just moving the files in there is a much easier thing to do.

Before reviewing, please briefly go through this Wiki: https://wiki.jmonkeyengine.org/doku.php/jme3:advanced:networking (you'll definately need it later on in the project anyway, so you might as well read it now).

So in order to create a server in the JME3 networking library, you need to call Network.createServer(port) to create a Server and Network.connectToServer(ip, port) to create a Client. Where should these two objects be stored?
That's where the ServerWrapper and ClientWrapper classes come in. Both have a default port (4321) stored and some useful methods. Mainly startServer, closeServer, startClient and closeClient.
Extra functionality could be added later on, but for now, these are the basic features of the classes.

The test suites of these classes are not that clean, but that had to be sacrificed to make the code testable. The main problem is that JME3 Networking doesn't like it when a server and a client run on the same instance. One of the more difficult things I had to do is in the TestClientWrapper. Using PowerMockito, I created a fake client using a different port than the default port. If you can't understand that part of the code, please leave a comment.

### CODE REVIEW

File: ContextApp/src/main/java/com/github/migi_1/ContextApp/ClientWrapper.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	
+	/**
+	 * Closes the Client.
+	 */
+	public void closeClient() {
+		if (client != null) {
+			client.close();
+			client = null;
+		} else {
+			throw new IllegalStateException("Client has already been started");
```
Comment (NilsHullegien) [inline ContextApp/src/main/java/com/github/migi_1/ContextApp/ClientWrapper.java]: I assume this should be: "Client has already been closed"?

File: ContextApp/src/test/java/com/github/migi_1/ContextApp/TestClientWrapper.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+		Client fakeClient = Network.connectToServer("localhost", fakePort);
+		
+		PowerMockito.mockStatic(Network.class);
+		
+		//Making sure that when a client is created, it connects to the fake port we are using.
+		//Otherwise bad things will happen during testing.
+		Mockito.when(Network.connectToServer("localhost", actualPort)).thenReturn(fakeClient);
+		
+		ClientWrapper client = ClientWrapper.getInstance();
+
```
Comment (NilsHullegien) [inline ContextApp/src/test/java/com/github/migi_1/ContextApp/TestClientWrapper.java:66]: Good idea of splitting this up into different methods

File: ContextDesktop/src/test/java/com/github/migi_1/Context/TestServerWrapper.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	 *   7. Asserts that the server is no longer running.
+	 *   8. Asserts that getServer returns null.
+	 *   9. Asserts that calling the closeServer method again gives an Exception.
+	 * 
+	 * @throws IOException 
+	 */
+	@Test
+	public void testServerWrapper() throws IOException {
+		ServerWrapper server = ServerWrapper.getInstance();
+
```
Comment (NilsHullegien) [inline ContextDesktop/src/test/java/com/github/migi_1/Context/TestServerWrapper.java:38]: Again, good idea to split this up into methods

File: ContextApp/src/main/java/com/github/migi_1/ContextApp/ClientWrapper.java (hunk 1 above)
Comment (LMiljak) [inline ContextApp/src/main/java/com/github/migi_1/ContextApp/ClientWrapper.java]: good catch. ill fix it now.

**Comment by NilsHullegien** [comment]
Looks good now!

**Comment by ThatFlint** [comment]
Looks a-okay

**Comment by LMiljak** [comment]
@ThatFlint if you're the second one to approve, you can also merge it :)

**Pull request merged**